invenio_communities/token.py

```diff
@@ -16,84 +16,59 @@
 from flask import current_app
 from itsdangerous import BadData, JSONWebSignatureSerializer, SignatureExpired
 
-SUPPORTED_DIGEST_ALGORITHMS = ('HS256', 'HS512')
 
+def create_token(obj_id, extra_data={}):
+    """Create a token referencing the object id with extra data.
 
-class TokenMixin(object):
-    """Mix-in class for token serializers."""
-
-    def create_token(self, obj_id, extra_data={}):
-        """Create a token referencing the object id with extra data.
-
-        Note random data is added to ensure that no two tokens are identical.
-        """
-        return self.dumps(
-            dict(
-                id=obj_id,
-                data=extra_data,
-                rnd=binascii.hexlify(os.urandom(4)).decode('utf-8')
-            )
+    Note random data is added to ensure that no two tokens are identical.
+    """
+    jss = JSONWebSignatureSerializer(current_app.config['SECRET_KEY'])
+    return jss.dumps(
+        dict(
+            id=obj_id,
+            data=extra_data,
+            rnd=binascii.hexlify(os.urandom(4)).decode('utf-8')
```

Comment:
Not sure if this is needed anymore.... Thus `expected_value` below can now check against the entire payload (instead of just the `data['id']`)